staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -492,16 +496,26 @@ func (p *patcher) applyPatch(_ context.Context, _, currentObject runtime.Object)
 	return objToUpdate, nil
 }
 
-func (p *patcher) admissionAttributes(ctx context.Context, updatedObject runtime.Object, currentObject runtime.Object, operation admission.Operation, operationOptions runtime.Object) admission.Attributes {
+func (p *patcher) admissionAttributes(ctx context.Context, updatedObject runtime.Object, currentObject runtime.Object, labels map[string]string, operation admission.Operation, operationOptions runtime.Object) admission.Attributes {
 	userInfo, _ := request.UserFrom(ctx)
-	return admission.NewAttributesRecord(updatedObject, currentObject, p.kind, p.namespace, p.name, p.resource, p.subresource, operation, operationOptions, p.dryRun, userInfo)
+	return admission.NewAttributesRecord(updatedObject, currentObject, labels, p.kind, p.namespace, p.name, p.resource, p.subresource, operation, operationOptions, p.dryRun, userInfo)
 }
 
 // applyAdmission is called every time GuaranteedUpdate asks for the updated object,
 // and is given the currently persisted object and the patched object as input.
 // TODO: rename this function because the name implies it is related to applyPatcher
 func (p *patcher) applyAdmission(ctx context.Context, patchedObject runtime.Object, currentObject runtime.Object) (runtime.Object, error) {
 	p.trace.Step("About to check admission control")
+	accessor, err := meta.Accessor(patchedObject)
+	if err != nil {
+		return nil, errors.NewInternalError(err)
+	}
+	labelsHolder, ok := request.LabelsHolderFrom(ctx)
+	if !ok {
+		debug.PrintStack()
+		return nil, errors.NewInternalError(fmt.Errorf("expect labelsHolder set in context"))
+	}
+	labelsHolder.SetLabels(accessor.GetLabels())
```

Comment:
The changes to the patch handler are complicated. I need to extract the labels after the patch is applied to the existing object, but before the running the mutation webhooks (`applyAdmission`), and pass the labels to the validation webhooks (encapsuled in `createValidation` and `updateValidation`). I used the `*ctx.LabelsHolder` to pass the labels.